crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -12,24 +12,53 @@ pub mod contracts;
 
 pub use self::contracts::Contracts;
 
+/// An Ethereum RPC connection.
+pub struct Rpc {
+    web3: DynWeb3,
+    network: Network,
+}
+
+/// Network information for an Ethereum blockchain connection.
+#[derive(Clone, Debug)]
+pub struct Network {
+    pub id: eth::NetworkId,
+    pub chain: eth::ChainId,
+}
+
+impl Rpc {
+    /// Instantiate an RPC connection to an Ethereum (or Ethereum-compatible)
+    /// node at the specifed URL.
+    pub async fn connect(url: &url::Url) -> Result<Self, Error> {
```

Comment:
Why not call this `new()`? It's not like calling this function will open a connection so the name seems misleading.

Reply:
It does though - specifically it uses the connection to fetch and cache the chain ID and network ID.